ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -225,7 +270,70 @@ def _address_family_evpn(self) -> list:
 
         return address_family_evpn
 
-    def _address_family_rtc(self) -> list | None:
+    def _address_family_ipv4_sr_te(self) -> dict | None:
+        """Generate structured config for IPv4 SR-TE address family"""
+        if not self.shared_utils.cv_pathfinder_role:
+            return None
+
+        address_family_ipv4_sr_te = {
+            "peer_groups": [
+                {
+                    "name": self.shared_utils.bgp_peer_groups["wan_overlay_peers"]["name"],
+                    "activate": True,
+                }
+            ],
+        }
+
+        return address_family_ipv4_sr_te
+
+    def _address_family_link_state(self) -> dict | None:
+        """Generate structured config for link-state address family"""
+        if not self.shared_utils.cv_pathfinder_role:
+            return None
+
+        address_family_link_state = {
+            "peer_groups": [
+                {
+                    "name": self.shared_utils.bgp_peer_groups["wan_overlay_peers"]["name"],
+                    "activate": True,
+                }
+            ],
+        }
+
+        if self.shared_utils.cv_pathfinder_role == "pathfinder":
+            address_family_link_state["path_selection"] = {"roles": {"consumer": True, "propagator": True}}
+            address_family_link_state["bgp"] = {
```

Comment:
This should be set on peer group instead of BGP since the RR peerings should not have this.